pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -329,13 +436,18 @@ func keepCurrentContainerResources(declared, current *appsv1.Deployment) bool {
 // deployment returns the deployment from the server
 func (r *reconcilerBase) deployment(ctx context.Context, dRef client.ObjectKey) (*appsv1.Deployment, error) {
 	depObj := &appsv1.Deployment{}
-	if err := r.client.Get(ctx, dRef, depObj); err != nil {
+	depUnObj, err := r.dynamicClient.Resource(kinds.DeploymentResource()).Namespace(dRef.Namespace).Get(ctx, dRef.Name, metav1.GetOptions{})
```

Comment:
Don't we already build this deploymentResourceInterface? Can it be passed in? If not, like elsewhere, the resource should be retrieved from the rest mapper, not hard-coded.